### PR TITLE
Switch to ament_cmake_vendor_package

### PR DESCRIPTION
**Pull request by cottsay**
Part of ament/ament_cmake#429

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18830/)
* Linux [Build Status](https://ci.ros2.org/job/ci_linux/18831/) (FORCE_BUILD_VENDOR_PKG)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13359/)
* Linux-rhel [Build Status](https://ci.ros2.org/job/ci_linux-rhel/421/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19575/)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,97 +1,25 @@
...
-    endif()
-    if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "^(x86_|x86-|AMD|amd|x)64$")
-      list(APPEND cmake_configure_args -A x64)
-    endif()
-  endif()
-
-  if(DEFINED CMAKE_BUILD_TYPE)
-    if(WIN32)
-      build_command(_build_command CONFIGURATION ${CMAKE_BUILD_TYPE})
-      list(APPEND cmake_commands "BUILD_COMMAND ${_build_command}")
```
Comment (Yadunund) [inline CMakeLists.txt:25]: Does `ament_vendor()` take care of this too? I can't tell from the implementation 🤔

**Review by Yadunund — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (cottsay) [reply CMakeLists.txt:25]: It doesn't. I'm unable to reproduce the warning that brought on that change, and I haven't seen that pattern in any other vendor packages.

Comment (Yadunund) [reply CMakeLists.txt:25]: Okay good to know 👍🏼

**Comment by Crola1702** [comment]
@cottsay @Yadunund was the CI run in this PR up to date? Today Rolling is failing with the following output:

```
--- stderr: mimick_vendor
CMake Error at /tmp/ws/install_isolated/ament_cmake_vendor_package/share/ament_cmake_vendor_package/cmake/ament_vendor.cmake:269 (find_program):
  Could not find vcs_EXECUTABLE using the following names: vcs
Call Stack (most recent call first):
  /tmp/ws/install_isolated/ament_cmake_vendor_package/share/ament_cmake_vendor_package/cmake/ament_vendor.cmake:107 (_ament_vendor)
  CMakeLists.txt:12 (ament_vendor)


---
Failed   <<< mimick_vendor [1.12s, exited with code 1]
```

Reference builds:
* [Rolling fastrtps](https://build.ros2.org/view/Rci/job/Rci__nightly-fastrtps_ubuntu_jammy_amd64/488/console)
* [Rolling CycloneDDS](https://build.ros2.org/view/Rci/job/Rci__nightly-cyclonedds_ubuntu_jammy_amd64/475/console)
* [Rolling Connext](https://build.ros2.org/view/Rci/job/Rci__nightly-connext_ubuntu_jammy_amd64/477/console)
* [Rolling Debug](https://build.ros2.org/view/Rci/job/Rci__nightly-debug_ubuntu_jammy_amd64/484/console)
* [Rolling Release](https://build.ros2.org/view/Rci/job/Rci__nightly-release_ubuntu_jammy_amd64/492/console)

**Comment by Yadunund** [comment]
Should be resolved by https://github.com/ament/ament_cmake/pull/462/files

**Pull request merged**